pkg/monitor/cluster/hiveregistrationstatus.go

```diff
@@ -31,20 +31,40 @@ func (mon *Monitor) emitHiveRegistrationStatus(ctx context.Context) error {
 		return fmt.Errorf("cluster %s not adopted. No namespace in the clusterdocument", mon.oc.Name)
 	}
 
+	return mon.validateHiveConditions(ctx)
+}
+
+func (mon *Monitor) validateHiveConditions(ctx context.Context) error {
```

Comment:
* Name is confusing: we do not perform any validation here. We filter conditions and emmit metrics.
* Also the name is quite generic. We work with `ClusterDeployment` conditions here.

Reply:
Renamed and simplified.